### ISSUE TITLE
Failed to load fw_payload.bin

### ISSUE THREAD

**Issue by Tiago-R** [openhwgroup/cva6-platform#9]
I've built the bitstream, fw_payload.bin and Linux image following the exact instructions given in this repo apart from using a single core ```protosyn -b genesys2 -d system -c ariane --x_tiles=1 --y_tiles=1 --linux_bootrom=uboot``` instead of ```protosyn -b genesys2 -d system -c ariane --x_tiles=2 --y_tiles=1 --linux_bootrom=uboot```. All compiles without errors.

I then format my SD card with an ext4 partition and filesystem and copy fw_payload.bin and Image to a /boot/ directory.

When booting the board I get an error when SPL tries to load /boot/fw_payload.bin:
```
Trying to boot from MMC1
spl: ext4fs_open failed
spl_load_image_ext: error reading image boot/fw_payload.bin, err - -1
SPL: failed to boot from all boot devices
### ERROR ### Please RESET the board ###
```
Resetting the board yields no different results.

**Comment by Jbalkind** [comment]
I'm glad to see someone is trying this! Have you tried the pre-built bitstream with your SD card too?

**Comment by Tiago-R** [comment]
Yes, in an attempt to isolate the problem I tried running the Fedora Summit image provided in both the precompiled bitstream and my compiled bitstream and both worked.

Given this, I think the problem can lie on how I am writing the SD card, as flashing the fedora image seems to working fine. 

Thank you for the help

**Comment by Jbalkind** [comment]
Ok I see. Could you explain a little more about how you put the ext4 on your SD? You did create a partition table before creating the ext4 fs, right?

**Comment by Tiago-R** [comment]
Of course.
I created the partition by running `sudo sgdisk --clear --new=1:2048:0 --typecode=1:8300 /dev/sdb` and then creating a file system with `mkfs.ext4 /dev/sdb1`. 
I then mounted the sdb1 partition, created the boot directory and copied the fw_payload.bin and Image files to it.

**Comment by Jbalkind** [comment]
Hm so issues I've seen in the past have been:
- simply SD corruption
- an ext4 fs that was somehow misconfigured but we weren't sure how. We could tell it apart because it had a different inode size
- for a while for the released image we were having issues where some of the images wouldn't work and we couldn't diagnose why (we were checking extended file permissions and a bunch of other stuff). I'm really trying to think of what the final fix ended up being but I'd need to dig back into our mattermost log

Does it work if you put your fw_payload.bin into the released filesystem image?

**Comment by Tiago-R** [comment]
I also tried that with and didn't work

**Comment by Jbalkind** [comment]
Can you check permissions after you do that? I've seen issues before where the file copied in without root and the correct rwx permissions.

**Comment by Tiago-R** [comment]
I don't have access to the board right now. When I do I will check that and let you know. Thanks for all the help so far

**Comment by Tiago-R** [comment]
Hello @Jbalkind 
I just now tried it again by giving all rwx permissions to fw_payload.bin with the `sudo chmod 777 fw_payload.bin` command.
The final result was the same error.

**Comment by Jbalkind** [comment]
Was the original fw_payload.bin permissioned at 777? I thought it had different permissions. Did you put the fw_payload.bin into the disk image and then dd? Or did you cp into the SD? I've been able to replace the `Image` inside the disk before. I don't exactly remember whether I could similarly do that with fw_payload.bin.

Might also be easier to discuss this on the openpiton zulip.

**Comment by Tiago-R** [comment]
I've sent you a dm on openpiton zulip server
